GraphBase/src/graph/DirectedGraph.java

```diff
@@ -0,0 +1,76 @@
+package graph;
+
+import java.util.Collection;
+import java.lang.Cloneable;
+
+public class DirectedGraph extends Graph implements Cloneable{
```

Comment:
space before {. You can use eclipse or intellij auto format to do this dummy thing for you.